### PR TITLE
Missing required lib

### LINKED ISSUE #47: implement missing_required_lib and dropping MySQL-python (Python 2.x only) at the same time?
I wonder if we should drop the MySQL-python (Python 2.x only) import check and Python 2.x documentation fragments?

https://github.com/ansible-collections/community.proxysql/blob/main/plugins/module_utils/mysql.py#L24

Imho [MySQL-python](https://github.com/farcepest/MySQLdb1) is dead. And PyMySQL supports at least Python 2.7 via distribution packages.

### PR DESCRIPTION
**Pull request by markuman**
##### SUMMARY
Rework error handling when required libraries are missing using `from ansible.module_utils.basic import missing_required_lib`

closes #47 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
Every module is affected in this collection

### CODE REVIEW

**Review by Andersson007 — COMMENTED** [review]
@markuman thanks for the PR!

File: plugins/module_utils/mysql.py [hunk 1]
```diff
@@ -15,20 +15,24 @@
...
     _mysql_cursor_param = 'cursor'
+    HAS_MYSQL_PACKAGE = True
 except ImportError:
     try:
         import MySQLdb as mysql_driver
         import MySQLdb.cursors
         _mysql_cursor_param = 'cursorclass'
+        HAS_MYSQL_PACKAGE = True
     except ImportError:
+        MYSQL_IMP_ERR = 'Cannot find PyMySQL (Python 2.7 and Python 3.X) or MySQL-python (Python 2.X) module.'
```
[the remark this answers is not in the document]
Comment (markuman) [reply plugins/module_utils/mysql.py:32]: Ah I missed that line.

> and it also supports Python3 


Yes, Pip/Github says "This project adds Python 3 support and bug fixes"  
and I also tested it with python3.

File: plugins/module_utils/mysql.py [hunk 2]
```diff
@@ -41,6 +45,9 @@ def mysql_connect(module, login_user=None, login_password=None, config_file='',
                   connect_timeout=30, autocommit=False, config_overrides_defaults=False):
     config = {}
 
+    if not HAS_MYSQL_PACKAGE:
+        module.fail_json(msg=missing_required_lib("PyMySQL"), exception=MYSQL_IMP_ERR)
```
Comment (markuman) [inline plugins/module_utils/mysql.py:49]: @Andersson007  I'm not sure about this line.  
Shall we keep only `PyMySQL` here? The self-written `MYSQL_IMP_ERR` includes both libraries.

Comment (Andersson007) [reply plugins/module_utils/mysql.py:49]: @markuman I'm also not sure. I wonder if we can use `PyMySQL|mysqlclient`  or something like that

**Comment by Andersson007** [comment]
We could also add a changelog fragment here with the `minor_changes:` key https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment saying, for example "- Refactoring of connector presence checking (https://github.com/ansible-collections/community.proxysql/pull/50)." or something

File: changelogs/fragments/50-refactor-connectors.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - Refactoring of connector presence checking (https://github.com/ansible-collections/community.proxysql/pull/50)
+  - replace MySQL-Python with mysqlclient (https://github.com/ansible-collections/community.proxysql/pull/50)
```
Comment (Andersson007) [inline changelogs/fragments/50-refactor-connectors.yaml:3]:
```suggestion
  - Refactoring of connector presence checking (https://github.com/ansible-collections/community.proxysql/pull/50).
  - Replace MySQL-Python with mysqlclient in the import error message (https://github.com/ansible-collections/community.proxysql/pull/50).
```

**Comment by Andersson007** [comment]
@markuman thanks for the refactoring!

**Pull request merged**